src/zc/buildout/__init__.py

```diff
@@ -13,9 +13,108 @@
 ##############################################################################
 """Buildout package
 """
+from setuptools.extern.six.moves import urllib
+from setuptools.package_index import PackageIndex
+from setuptools.package_index import URL_SCHEME
+from setuptools.package_index import HREF
+from setuptools.package_index import distros_for_url
+from setuptools.package_index import htmldecode
+
+from pip._internal.index.collector import HTMLPage
+from pip._internal.index.collector import parse_links
+from pip._internal.index.package_finder import _check_link_requires_python
+from pip._internal.models.target_python import TargetPython
+from pip._vendor import six
```

Comment:
Oh no.  Oh no no no.  pip does not want anyone to be relying on pip's internals.  This is going to be an unmaintainable nightmare.

Reply:
I totally agree this could be a big risk.
This is only needed for PEP345 support. This is one of the questions I wanted to tackle with the buildout community. Would we vendor include what is neded by `parse_links`, `_check_link_requires_python` and `TargetPython`.

Reply:
From what I've seen on distutils-sig@, the pip developers are open to people extracting bits of reusable logic from pip into standalone libraries like `packaging` or `distlib` or `pep517` (that they then will vendor).  That would be the most future-proof way of sharing pip's logic.

It might be a good idea to discuss this on distutils-sig@.

Reply:
As a first step, I made it less fragile by making it optional and warning if it fails.